src/Listeners/CopyModels.php

```diff
@@ -19,27 +23,53 @@ public function __construct(
     public function handle(TableCreated $event)
     {
         if ($event->model === null) {
-            throw SchemaModelException::create($event->table);
+            throw SchemaModelException::wrongCreateMethod($event->table);
         }
 
         if ($event->version === null) {
             return;
         }
 
+        $active = $this->versions->active();
+
         $this->versions->clearActive();
 
         /** @var Model&Versioned $model */
         $model = new ($event->model);
         $models = $model->newQueryWithoutScopes()->cursor();
         $this->versions->setActive($event->version);
 
-        Schema::withoutForeignKeyConstraints(function () use ($models) {
+        $this->quietlyReplicate($event->model, function () use ($models) {
             $models->each(function (Model&Versioned $model) {
                 $id = $model->getKeyName();
                 $replicated = $model->replicate();
                 $replicated->$id = $model->$id;
+                $replicated->setCreatedAt($model->{$model->getCreatedAtColumn()});
+                $replicated->setUpdatedAt($model->{$model->getUpdatedAtColumn()});
                 $replicated->save();
             });
         });
+
+        $this->versions->setActive($active);
+
+        Event::dispatch(TableCopied::fromCreated($event));
+    }
+
+    /**
+     * @var class-string<Model&Versioned>
+     */
+    protected function quietlyReplicate(string $model, Closure $callback)
+    {
+        Schema::disableForeignKeyConstraints();
+
+        $model::withoutTimestamps(function () use ($model, $callback) {
+            if (config('snapshots.history') && in_array(HushesHandlers::class, class_uses_recursive($model))) {
```

Comment:
maybe this to match the soft delete check, for readability?
```suggestion
            $hushed = config('snapshots.history') && in_array(HushesHandlers::class, class_uses_recursive($model))
            if ($hushed) {
```

or if you can think of a more verbose variable name

Reply:
I generally only extract it when it is referenced more than once and needed multiple times.

I actually don't do it for readability, and my personal preference for readability is not to create unnecessary temp vars. But I don't have strong feelings about that. We can maybe look into adding some Laravel coding standards we can all reference. What is most important to me is we are all consistent, whatever standards we choose.